llama_index/agent/react/base.py

```diff
@@ -76,36 +73,25 @@ def __init__(
     def from_tools(
         cls,
         tools: Optional[List[BaseTool]] = None,
-        tool_retriever: Optional[ObjectRetriever[BaseTool]] = None,
         llm: Optional[LLM] = None,
         chat_history: Optional[List[ChatMessage]] = None,
         memory: Optional[BaseMemory] = None,
         memory_cls: Type[BaseMemory] = ChatMemoryBuffer,
-        max_iterations: int = 10,
-        react_chat_formatter: Optional[ReActChatFormatter] = None,
-        output_parser: Optional[ReActOutputParser] = None,
         callback_manager: Optional[CallbackManager] = None,
-        verbose: bool = False,
         **kwargs: Any,
     ) -> "ReActAgent":
-        tools = tools or []
-        chat_history = chat_history or []
         llm = llm or OpenAI(model=DEFAULT_MODEL_NAME)
         if callback_manager is not None:
             llm.callback_manager = callback_manager
-
-        memory = memory or memory_cls.from_defaults(chat_history=chat_history, llm=llm)
-
+        memory = memory or memory_cls.from_defaults(
+            chat_history=chat_history or [], llm=llm
+        )
         return cls(
-            tools=tools,
-            tool_retriever=tool_retriever,
+            tools=tools or [],
             llm=llm,
             memory=memory,
-            max_iterations=max_iterations,
-            react_chat_formatter=react_chat_formatter,
-            output_parser=output_parser,
```

Comment:
I think I disagree with removing all these arguments? If we rely on kwargs now, it doesn't show up in the IDE autocomplete 🤔

Reply:
Yeah that's a valid argument.  I guess in my own style, I prefer to duplicate names/typing/defaults less.

Imo it's more of an issue with autocomplete not being able to follow `**kwargs` than the source code.  

I will just bring back the complete signature!